pkg/controllers/controller.go

```diff
@@ -0,0 +1,6 @@
+package controllers
+
+type Controller interface {
+	// Run starts the controller
+	Run(stopCh <-chan struct{})
```

Comment:
in `main`, line 448.

Reply:
Ah, it's just an interface, I thought about using it in `main` later.

Reply:
It would be awesome.

Reply:
I started working on the config controller.

Reply:
A BIG improvement if we can get this done!

Reply:
Yeah, it will make the code easier to read and with better separation of concerns.
Our controllers also need some cleanup, they rely too much on event handlers, we need more reconciliation oriented logic.